pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -204,37 +203,15 @@ func TestReconcile(t *testing.T) {
 	// after we have resolved them.
 	rs[0].SelfLink = "some/link"
 
-	d := test.Data{
-		PipelineRuns:      prs,
-		Pipelines:         ps,
-		Tasks:             ts,
-		ClusterTasks:      clusterTasks,
-		PipelineResources: rs,
-	}
-
-	testAssets, cancel := getPipelineRunController(t, d)
-	defer cancel()
-	c := testAssets.Controller
-	clients := testAssets.Clients
-
-	if err := c.Reconciler.Reconcile(context.Background(), "foo/test-pipeline-run-success"); err != nil {
-		t.Fatalf("Error reconciling: %s", err)
-	}
-
-	if len(clients.Pipeline.Actions()) == 0 {
-		t.Fatalf("Expected client to have been used to create a TaskRun but it wasn't")
-	}
-
-	t.Log("actions", clients.Pipeline.Actions())
+	reconciledRun, clients := getReconciledRun(t, prs, ps, ts, clusterTasks, nil, rs, nil, "foo", "test-pipeline-run-success")
+	actions := clients.Pipeline.Actions()
 
-	// Check that the PipelineRun was reconciled correctly
-	reconciledRun, err := clients.Pipeline.TektonV1beta1().PipelineRuns("foo").Get("test-pipeline-run-success", metav1.GetOptions{})
-	if err != nil {
-		t.Fatalf("Somehow had error getting reconciled run out of fake client: %s", err)
+	if len(actions) < 2 {
+		t.Fatalf("Expected client to have at least two action implementation but it has %d", len(actions))
```

Comment:
Have added check on length before accessing that index in the list

Reply:
thanks for explaining! really helps when reviewing :D

and good call!!